notificationworkerlambda/src/main/scala/com/gu/notifications/worker/AndroidSender.scala

```diff
@@ -37,26 +37,24 @@ class AndroidSender(val config: FcmWorkerConfiguration, val firebaseAppName: Opt
 
   //override the deliverChunkedTokens method to validate the success of sending batch notifications to the FCM client. This implementation could be refactored in the future to make it more streamlined with APNs
   override def deliverChunkedTokens(chunkedTokenStream: Stream[IO, (ChunkedTokens, Long, Instant)]): Stream[IO, Unit] = {
-    for {
-      (chunkedTokens, sentTime, functionStartTime) <- chunkedTokenStream
-      isAllowedToSendBatch = chunkedTokens.notification.topic.forall(topic => config.allowedTopicsForBatchSend.contains(topic.toString))
-      resp <- {
+    chunkedTokenStream.map {
+      case (chunkedTokens, sentTime, functionStartTime) =>
+        val isAllowedToSendBatch = chunkedTokens.notification.topic.forall(topic => config.allowedTopicsForBatchSend.contains(topic.toString))
+
         if (isAllowedToSendBatch) {
           logger.info(s"Allowed to send notification ${chunkedTokens.notification.id} to topics ${chunkedTokens.notification.topic} in batches")
           deliverBatchNotificationStream(Stream.emits(chunkedTokens.toBatchNotificationToSends).covary[IO])
             .broadcastTo(
               reportBatchSuccesses(chunkedTokens, sentTime, functionStartTime),
               cleanupBatchFailures(chunkedTokens.notification.id),
               trackBatchProgress(chunkedTokens.notification.id))
-        }
-        else
+        } else
           deliverIndividualNotificationStream(Stream.emits(chunkedTokens.toNotificationToSends).covary[IO])
             .broadcastTo(
               reportSuccesses(chunkedTokens, sentTime, functionStartTime),
               cleanupFailures,
               trackProgress(chunkedTokens.notification.id))
-      }
-    } yield resp
+    }.parJoin(maxConcurrency)
```

Comment:
This is a change that becomes necessary as we increase the SQS batchSize of records above 1 : we don't want to process those records (each containing at most 1000 device ids) one after the other, we want to process them in parallel!

The [`parJoin()`](https://github.com/typelevel/fs2/blob/86e27ca52fc5767c066b91b55d30357764ad8401/core/shared/src/main/scala/fs2/Stream.scala#L4014-L4044) method merges a stream of streams (the stream of records, each record yielding a stream of tokens) and runs multiple record streams in parallel, rather than all sequentially. We already use `parJoin` in several places, eg:

https://github.com/guardian/mobile-n10n/blob/07259d294e34b184c41ecf34a0a4785e79d11332/notificationworkerlambda/src/main/scala/com/gu/notifications/worker/SenderRequestHandler.scala#L64-L67

Note that the change in this file only affects the `AndroidSender`, which thanks to #747 already has a custom overridden `deliverChunkedTokens()` method to allow it to batch requests. To keep iOS in line with the new `parJoin()` behaviour, the original method in the `SenderRequestHandler` base trait has also been updated.